spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1026,8 +1024,21 @@ def document_did_open(self):
         }
         return params
 
-    # ------------- LSP: Linting ---------------------------------------
+    # ------------- LSP: Symbols ---------------------------------------
+    @request(method=LSPRequestTypes.DOCUMENT_SYMBOL)
+    def request_symbols(self):
+        """Trigger document symbols."""
+        params = {'file': self.filename}
+        return params
+
+    @handles(LSPRequestTypes.DOCUMENT_SYMBOL)
+    def process_symbols(self, params):
+        """Handle symbols response."""
+        symbols = params['params']
+        if symbols:
+            self.classfuncdropdown.update_data(symbols)

```

Comment:
Please put the code of this method inside a try/except like this:

```python
try:
    symbols = params['params']
    if symbols:
        self.classfuncdropdown.update_data(symbols)
except RuntimeError:
    # This is triggered when a codeeditor instance was removed
    # before the response can be processed.
    return
except Exception:
    self.log_lsp_handle_errors("Error when processing symbols")
```

This will correctly handle errors when processing symbols.